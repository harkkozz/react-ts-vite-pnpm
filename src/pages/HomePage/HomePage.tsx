import * as stylex from '@stylexjs/stylex';
import { useNavigate } from 'react-router';

import Button from 'components/Button/Button';

const styles = stylex.create({
  heading: {
    textAlign: 'center',
  },
});

const HomePage = () => {
  const navigate = useNavigate();

  return (
    <div>
      <h2 {...stylex.props(styles.heading)}>Main page</h2>
      <Button text="Other page" onClick={() => navigate('other-page')} bgColor="red" />
    </div>
  );
};

export default HomePage;
