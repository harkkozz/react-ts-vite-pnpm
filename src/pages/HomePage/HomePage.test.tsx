import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import Test from 'testUtils/TestApp';

import HomePage from './HomePage';

const onUseNavigate = vi.fn();

describe('Home page', () => {
  it('Should match snapshot', () => {
    const { baseElement } = render(
      <Test>
        <HomePage />
      </Test>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('Should navigate to other page', () => {
    const { getAllByRole } = render(
      <Test>
        <HomePage />
      </Test>,
    );
    const onNextPage = getAllByRole('button', { name: 'Other page' });
    fireEvent.click(onNextPage[0]);
    expect(onUseNavigate).toHaveBeenCalledTimes(0);
  });
});
