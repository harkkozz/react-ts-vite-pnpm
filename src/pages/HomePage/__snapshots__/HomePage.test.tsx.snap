// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > Should match snapshot 1`] = `
<body>
  <div>
    <div>
      <h2
        class="HomePage__styles.heading x2b8uid"
      >
        Main page
      </h2>
      <button
        class="Button__styles.button x1ypdohk x78zum5 x1qihwyb xmgby6i xddr76m x1835gkm x15tramd xq1mx2j xkpwil5"
        style="--backgroundColor: red;"
      >
        Other page
      </button>
    </div>
  </div>
</body>
`;
